src/charm.py

```diff
@@ -145,24 +175,44 @@ def on_update_status(self, event):
     #        PEER RELATION HOOK HANDLERS         #
     ##############################################
 
-    # hooks: cluster-relation-changed, cluster-relation-departed
+    # Handles relation-changed and relation-departed events
     def reconfigure(self, event):
+        """Reconfigure replicat set
+
+        The number of replicas in the MongoDB replica set is updated.
+        """
         logger.debug("Running reconfigure")
 
         if (
             self.unit.is_leader()
-            and self.state.mongodb_initialized
             and self.need_replica_set_reconfiguration()
         ):
-            self.mongo.reconfigure_replica_set(self.cluster_hosts)
+            try:
+                self.mongo.reconfigure_replica_set(self.cluster_hosts)
+            except Exception as e:
+                logger.info("Deferring relation event since : error={}".format(e))
+                event.defer()
+
         self.on_update_status(event)
         logger.debug("Running reconfigure finished")
 
     ##############################################
     #               RELATIONS                    #
     ##############################################
 
+    # handles client relation for MongoDB
     def on_database_relation_changed(self, event):
+        """Connect to database client
+
+        Any MongoDB client is provided with the following information
+        - Is MongoDB in a replicated or unitary state
+        - Replica set URI
+        - Standalone URI
```

Comment:
Is this true anymore? I thought there was no longer a standalone deployment.

Reply:
There is not standalone deployment configuation option, but the user can always choose to deploy one unit and effectively be in standalone mode. I did not remove the standalone URI to allow a client charm wants to use it to access a single unit of MongoDB.

Reply:
Ah I see. I was assuming the standalone URI wouldn't work if mongo was a replica set.